tests/clients/sendgrid/test_sendgrid.py

```diff
@@ -0,0 +1,108 @@
+import os
+import string
+import gevent
+import imaplib
+import email
+from unittest import TestCase
+from jumpscale.loader import j
+
+
+def generate_rand_text(char_count, choices):
+    return j.data.idgenerator.nfromchoices(10, string.ascii_letters)
+
+
+class Sendgrid(TestCase):
+    def setUp(self):
+        self.sendgird_client_name = generate_rand_text(10, string.ascii_letters)
+        self.test = j.clients.sendgrid.get(name=self.sendgird_client_name)
+        if os.getenv("SEND_GRID_API_KEY_TOKEN") and os.getenv("RECIPIENT_MAIL"):
```

Comment:
Instead of getting env vars twice, we can get it once in the variables and then check the variables.